diff -up libEMF-1.0.6/include/libEMF/wine/winnt.h.amd64 libEMF-1.0.6/include/libEMF/wine/winnt.h
--- libEMF-1.0.6/include/libEMF/wine/winnt.h.amd64	2012-05-24 17:52:51.000000000 +0200
+++ libEMF-1.0.6/include/libEMF/wine/winnt.h	2012-09-03 22:49:49.000000000 +0200
@@ -37,6 +37,10 @@
 # undef  WORDS_BIGENDIAN
 # undef  BITFIELDS_BIGENDIAN
 # define ALLOW_UNALIGNED_ACCESS
+#elif defined(__alpha__)
+# undef  WORDS_BIGENDIAN
+# undef  BITFIELDS_BIGENDIAN
+# undef  ALLOW_UNALIGNED_ACCESS
 #elif defined(__sparc__)
 # define WORDS_BIGENDIAN
 # define BITFIELDS_BIGENDIAN
@@ -238,7 +242,7 @@ typedef unsigned short  WORD,       *PWO
 typedef int             INT,        *PINT,     *LPINT;
 typedef unsigned int    UINT,       *PUINT,    *LPUINT;
 /* Not sure this is correct. Probably should depend on the compiler, too. */
-#if defined( __x86_64__)
+#if defined( __x86_64__) || defined(__alpha__)
 typedef unsigned int   DWORD,      *PDWORD,   *LPDWORD;
 typedef unsigned int   ULONG,      *PULONG,   *LPULONG;
 #else
@@ -284,7 +288,7 @@ typedef VOID           *PVOID,      *LPV
 typedef BYTE            BOOLEAN,    *PBOOLEAN;
 typedef char            CHAR,       *PCHAR;
 typedef short           SHORT,      *PSHORT;
-#if defined(__x86_64__)
+#if defined(__x86_64__) || defined(__alpha__)
 typedef int             LONG,       *PLONG,    *LPLONG;
 #else
 typedef long            LONG,       *PLONG,    *LPLONG;
